core/dbt/task/base.py

```diff
@@ -167,12 +168,31 @@ def move_to_nearest_project_dir(project_dir: Optional[str]) -> str:
     return nearest_project_dir
 
 
+# TODO: look into deprecating this class in favor of several small functions that
+# produce the same behavior. currently this class only contains manifest compilation,
+# holding a manifest, and moving direcories.
 class ConfiguredTask(BaseTask):
     ConfigType = RuntimeConfig
 
-    def __init__(self, args, config):
+    def __init__(self, args, config, manifest: Optional[Manifest]):
         super().__init__(args, config)
-        register_adapter(self.config)
+        self.graph: Optional[Graph] = None
+        self.manifest = manifest
+
+    def compile_manifest(self):
+        if self.manifest is None:
+            raise InternalException("compile_manifest called before manifest was loaded")
+
+        start_compile_manifest = time.perf_counter()
+
+        # we cannot get adapter in init since it will break rpc #5579
+        adapter = get_adapter(self.config)
```

Comment:
Was this because adapter haven't been initialized at the time? I kinda forgot about the change

Reply:
Compilation requires access to the adapter. Way back in https://github.com/dbt-labs/dbt-core/pull/2712, we made it possible for adapters to:
- Change the prefix used for ephemeral model CTEs [here](https://github.com/dbt-labs/dbt-core/blob/93619a9a3761d3606d257c1274828ac4eef4f627/core/dbt/compilation.py#L187-L190), since certain data platforms (e.g. Exasol) have very particular rules about which naming conventions are supported
- [Totally reimplement the `Compiler`](https://github.com/dbt-labs/dbt-core/blob/93619a9a3761d3606d257c1274828ac4eef4f627/core/dbt/adapters/base/impl.py#L1117-L1120), from scratch—intended for cases where an adapter needs to do something fundamentally different for ephemeral models, because the data platform doesn't support nested CTEs (e.g. SQLServer / Azure Synapse). But the [`Compiler`](https://github.com/dbt-labs/dbt-core/blob/93619a9a3761d3606d257c1274828ac4eef4f627/core/dbt/compilation.py#L161) also defines methods related to DAG construction (node/edge linking), which really should _not_ vary by data platform.

Compilation is part of execution/runtime, so requiring adapter-specific inputs for certain bits of its functionality can be appropriate. But I'd say that both of these are pretty wonky behaviors today, and are worth additional consideration as we move toward a future where the adapter is no longer a global singleton.